### PR TITLE
[OYPD-220] Membership registration display

### PR DESCRIPTION
**Pull request by david-hernandez**
Updates for membership form.

- [x] Creating landing page.
- [x] Add membership calculator paragraph.
- [x] Check the display through the three steps of the form.

screen shot 2017-02-19 at 8 16 56 pm
screen shot 2017-02-19 at 8 17 43 pm
screen shot 2017-02-19 at 8 17 56 pm
screen shot 2017-02-19 at 8 18 09 pm

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/499/

**Build Log**
*last 25 lines*

 ```
[...truncated 248 lines...]
stdout: You are about to create a /var/www/build499/sites/default/settings.php file and CREATE the 'drupal499' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=12   changed=4    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by david-hernandez** [comment]
I want to make sure this generally works. One thing that needs to be fixed is making the membership type boxes are equal height. I didn't do that yet because that is not straight forward. I have get access to the fieldset in order to add the bootstrap equal height classes, but a suggestion for that does not exist. A custom one would need to be made.

**Comment by alexschedrov** [comment]
```
01:30:32 TASK: [Installing drupal] ***************************************************** 
01:31:23 failed: [localhost] => {"changed": true, "cmd": "APP_ENV=dev drush -y si openy --db-url=mysql://root:root@127.0.0.1:/drupal499 --account-name=admin --account-pass=openy --uri=http://openy-dev.ffwua.com/build499", "delta": "0:00:51.614685", "end": "2017-02-20 01:31:23.734824", "rc": 1, "start": "2017-02-20 01:30:32.120139", "warnings": []}
01:31:23 stderr: Starting Drupal installation. This takes a while. Consider using the [ok]
01:31:23 --notify global option.
01:31:23 Drupal\Core\Config\UnmetDependenciesException: Configuration objects [error]
01:31:23 (core.entity_view_display.paragraph.membership_info.preview) provided
01:31:23 by openy_loc_branch have unmet dependencies in
01:31:23 /var/www/build499/core/lib/Drupal/Core/Config/UnmetDependenciesException.php:84
01:31:23 Stack trace:
01:31:23 #0
01:31:23 /var/www/build499/core/lib/Drupal/Core/Config/ConfigInstaller.php(460):
01:31:23 Drupal\Core\Config\UnmetDependenciesException::create('openy_loc_branc...',
01:31:23 Array)
01:31:23 #1
01:31:23 /var/www/build499/core/lib/Drupal/Core/Extension/ModuleInstaller.php(141):
01:31:23 Drupal\Core\Config\ConfigInstaller->checkConfigurationToInstall('module',
01:31:23 'openy_loc_branc...')
```

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/508/

**Build Log**
*last 25 lines*

 ```
[...truncated 248 lines...]
stdout: You are about to create a /var/www/build508/sites/default/settings.php file and CREATE the 'drupal508' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=12   changed=4    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/509/

**Build Log**
*last 25 lines*

 ```
[...truncated 248 lines...]
stdout: You are about to create a /var/www/build509/sites/default/settings.php file and CREATE the 'drupal509' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=12   changed=4    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build514

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/514/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build514/profiles/openy/build/reports/behat

File: themes/openy_themes/openy_rose/templates/custom/openy-calc-form-summary.html.twig [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<div class="calc-summary">
+  <div class="row">
+    <div class="col-md-6 col-sm-12 selected-branch calc-summary-col">
+      <h2>{{ 'Your selected branch'|t }}</h2>
+      <div>{{ result.location }}</div>
+    </div>
+    <div class="col-md-6 col-sm-12 selected-membership calc-summary-col">
+      <h2>{{ 'Your selected membership'|t }}</h2>
+      <div>{{ result.membership }}</div>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/custom/openy-calc-form-summary.html.twig:9]: According to task spec we should render "Calculator summary view mode" here.

File: themes/openy_themes/openy_rose/templates/form/form-element-membership-type.html.twig [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      {{ description.content }}
+    </div>
+  {% endif %}
+  {{ children }}
+  {% if suffix is not empty %}
+    <span class="field-suffix">{{ suffix }}</span>
+  {% endif %}
+  {% if label_display == 'after' %}
+    <div class="form-element-wrapper">
+      <h3>{{ data.title }}</h3>
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/form/form-element-membership-type.html.twig:92]: According to task spec we should render "Calculator preview view mode" here for membership type.

File: modules/openy_features/openy_location/modules/openy_loc_branch/openy_loc_branch.features.yml [hunk 3]
```diff
@@ -1 +1,3 @@
+excluded:
```
Comment (alexschedrov) [inline modules/openy_features/openy_location/modules/openy_loc_branch/openy_loc_branch.features.yml]: Can we get rid of this? I don't see `core.entity_view_display.paragraph.membership_info.preview` in the configs for some reasons.

File: modules/openy_features/openy_node/modules/openy_node_mbrshp/openy_node_mbrshp.features.yml [hunk 4]
```diff
@@ -1 +1,3 @@
+excluded:
+  - core.entity_view_display.paragraph.membership_info.default
```
Comment (alexschedrov) [inline modules/openy_features/openy_node/modules/openy_node_mbrshp/openy_node_mbrshp.features.yml]: Can we get rid of this? I don't see `core.entity_view_display.paragraph.membership_info.default` in the configs.

File: modules/openy_features/openy_prgf/openy_prgf.features.yml [hunk 5]
```diff
@@ -1 +1,3 @@
+excluded:
+  - core.entity_view_mode.paragraph.preview
```
Comment (alexschedrov) [inline modules/openy_features/openy_prgf/openy_prgf.features.yml]: Can we get rid of this? We should have this preview mode in configs.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
1. Please remove prices from this screen
image
2. Please remove description from this screen image

3. Center button
image

**Comment by david-hernandez** [comment]
In the last commit I didn't center the button. Can that be left until the completion button has an actual link and functions? Right now it is just a single div. I would have to force a width on it to center it or add some messy CSS that will probably get removed later. Once it has an actual link, meaning a div with another tag, it will be easy to center.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build525

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/525/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build525/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build535

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/535/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build535/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved. Treated as MVP, but needs some polishing.

**Pull request merged**